app/src/org/commcare/android/tasks/ResourceEngineTask.java

```diff
@@ -230,10 +228,6 @@ protected ResourceEngineOutcomes doTaskBackground(String... profileRefs) {
                 platform.upgrade(global, temporary, recovery);
 
                 // And see where we ended up to see whether an upgrade actually occurred
-            } else if (partialMode) {

```

Comment:
partialMode was always false, so let's just remove this, especially since the 'else' body below has pretty much the same behaviour